src/app/shared/store/chat.state.ts

```diff
@@ -49,12 +52,15 @@ export class ChatState {
     return state.selectedChatRoomMessages;
   }
 
-  constructor(private chatService: ChatService) {}
+  constructor(private chatService: ChatService, private store: Store) {}
 
   @Action(GetUserChatRooms)
-  getUserChatRooms({ patchState }: StateContext<ChatStateModel>, { userRole }: GetUserChatRooms): Observable<ChatRoom[]> {
+  getUserChatRooms({ patchState }: StateContext<ChatStateModel>, { parameters }: GetUserChatRooms): Observable<SearchResponse<ChatRoom[]>> {
+    const role = this.store.selectSnapshot(RegistrationState.role);
```

Comment:
okay, I see there was no Store in this state as well
co, can we pass the role as a parameter (added role to the interface for chatroom params) from the component? I believe in the component you already know the role, so you can add it as a param

as a result we get rid of logic regarding detecting role twice and get rid of store in the chat state